src/webapp/components/org-unit-selector/OrgUnitSelector.tsx

```diff
@@ -36,57 +37,38 @@ export const OrgUnitSelector: React.FC<OrgUnitSelectorProps> = React.memo(props
     const { onChange, selected, disabled } = props;
     const { api, currentUser, compositionRoot } = useAppContext();
 
-    const classes = useStyles();
+    const theme = useTheme();
     const snackbar = useSnackbar();
+    const classes = useStyles();
 
     const [isOpen, { enable: open, disable: close }] = useBooleanState(false);
     const [loading, { enable: startLoading, disable: stopLoading }] = useBooleanState(false);
-    const [onlyUserOU, setOnlyUserOU] = React.useState(false);
-    const [current, setCurrent] = React.useState<{ selectedPaths: string[]; onlyUserOU: boolean }>({
-        selectedPaths: [],
-        onlyUserOU: false,
-    });
+    const [currentPaths, setCurrentPaths] = React.useState<string[]>([]);
 
     const selectedPaths = React.useMemo(() => {
         const paths = selected.map(({ path }) => path);
         const deepestCommonRoot = findCommonRoot(paths);
         return deepestCommonRoot === "" ? [] : [deepestCommonRoot];
     }, [selected]);
 
-    const toggleUserDataSetsOnly = React.useCallback(
-        (e: React.ChangeEvent<HTMLInputElement>) => {
-            setCurrent(({ selectedPaths }) => ({
-                selectedPaths: selectedPaths,
-                onlyUserOU: e.target.checked,
-            }));
-        },
-        [setCurrent]
-    );
-
-    const updateSelected = React.useCallback((paths: string[]) => {
-        setCurrent(({ onlyUserOU }) => ({ selectedPaths: paths, onlyUserOU: onlyUserOU }));
-    }, []);
-
     const cancel = React.useCallback(() => {
-        setCurrent({ selectedPaths: selectedPaths, onlyUserOU: onlyUserOU });
+        setCurrentPaths(selectedPaths);
         close();
-    }, [close, onlyUserOU, selectedPaths]);
+    }, [close, selectedPaths]);
 
     const apply = React.useCallback(() => {
         startLoading();
-        const userOrgUnitIds = currentUser.organisationUnits.map(getId);
-        const orgUnitIds = _(current.selectedPaths)
+
+        const orgUnitIds = _(currentPaths)
             .map(path => path.split("/").slice(-1))
             .flatten()
             .value();
 
-        const ids = current.onlyUserOU ? userOrgUnitIds : orgUnitIds;
-        const getOrgUnits$ = compositionRoot.orgUnits.getWithChildren.execute(ids);
+        const getOrgUnits$ = compositionRoot.orgUnits.getWithChildren.execute(orgUnitIds);
```

Comment:
The correct way (to get cancellation) is to run this as an useEffect. You can do it youself (useCallback + useState + useEffect) or check `useCallbackEffect` from some EST project.

Reply:
Thxx!! In this case I don't need the cancellation function only just run, should I use `useCallbackEffect` anyways? @tokland